ui/src/groups/AddGroupSheet.tsx

```diff
@@ -0,0 +1,84 @@
+import { useCallback, useState } from 'react';
+import cn from 'classnames';
+import WidgetDrawer from '@/components/WidgetDrawer';
+import HomeIconMobileNav from '@/components/icons/HomeIconMobileNav';
+import NewRaysIcon from '@/components/icons/NewRaysIcon';
+import { isNativeApp } from '@/logic/native';
+import JoinGroup from './AddGroupSheet/JoinGroup';
+import CreateGroup from './AddGroupSheet/CreateGroup';
+
+function DragHandle() {
+  return (
+    <div className="my-3 flex w-full justify-center">
+      <div className="h-[5px] w-[32px] rounded-[100px] bg-gray-100" />
+    </div>
+  );
+}
+
+export default function AddGroupSheet(props: {
+  open: boolean;
+  onOpenChange: (open: boolean) => void;
+}) {
+  const [view, setView] = useState<'root' | 'create' | 'join'>('root');
+
+  const onOpenChange = (open: boolean) => {
+    if (!open) {
+      setView('root');
+    }
+    if (props.onOpenChange) {
+      props.onOpenChange(open);
+    }
+  };
+
+  const CreateOrJoin = useCallback(() => {
+    return (
```

Comment:
ESLint is warning about this - you can either directly return the component, or eliminate the var altogether and put the JSX code into the main component return statement below

```jsx
const CreateOrJoin = useCallback(() => (
  <div ... />
), []);
```